src/Security/AuthenticationSuccessHandler.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+/*
+ * This file is part of the package t3g/symfony-ldap-bundle.
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace T3G\Bundle\LdapBundle\Security;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Http\Authentication\DefaultAuthenticationSuccessHandler;
+use Symfony\Component\Security\Http\HttpUtils;
+
+class AuthenticationSuccessHandler extends DefaultAuthenticationSuccessHandler
+{
+    /**
+     * @var FlashBagInterface
+     */
+    private $flashBag;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param HttpUtils $httpUtils
+     * @param array $options
+     * @param FlashBagInterface|null $flashBag
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct(
+        HttpUtils $httpUtils,
+        array $options = [],
+        FlashBagInterface $flashBag = null,
+        LoggerInterface $logger = null
```

Comment:
Why and when can those be null?

Reply:
I think this was done by Lolli when he created this handler. It seems the only purpose of this `null` is to supress `optional parameter before required parameter`. I'll remove it. It seems to not impact functionality.

Reply:
What we can easily do here is to shift the parameters. The interface does not define how the constructor should look like and since we override the constructor we can freely arrange the parameters and put the optional one at the end.